tests/test.py

```diff
@@ -10138,6 +10138,211 @@ class TestPostgresqlTypeAwareDecryptionWithDefaultsPsycopg3(Psycopg3ExecutorMixi
+
+        # generate random id
+        id = get_random_id()
+
+        def check():
+            # ensure it is encrypted (if should_match) or ensure it's not encrypted (if not should_match)
+            if should_match:
+                # ensure decrypted data matches what was inserted
+
+                result = self.engine1.execute(f"SELECT {column_name} FROM {table_name} WHERE id={id};")
+                row = result.fetchone()
+
+                if bytes(row[row_name]) != bytes(test_string, "UTF-8"):
+                    self.fail(f"Table identifier {table_name}, column identifier {column_name}, did not match (decryption failed?), {bytes(row['data'])} != {test_string}")
+
+                # ensure database does not contain plaintext
+
+                result = self.engine2.execute(f"SELECT {column_name} FROM {table_name} WHERE id={id};")
+                row = result.fetchone()
+
+                if bytes(row[row_name]) == bytes(test_string, "UTF-8"):
+                    self.fail(f"Table identifier {table_name}, column identifier {column_name}, did not match (DB contains plaintext), {bytes(row['data'])} == {test_string}")
+
+                result = self.engine_raw.execute(f"SELECT {column_name} FROM {table_name} WHERE id={id};")
+                row = result.fetchone()
+
+                if bytes(row[row_name]) == bytes(test_string, "UTF-8"):
+                    self.fail(f"Table identifier {table_name}, column identifier {column_name}, did not match (DB contains plaintext), {bytes(row['data'])} == {test_string}")
+            else:
+                # ensure database contains plaintext
+
+                result = self.engine2.execute(f"SELECT {column_name} FROM {table_name} WHERE id={id};")
+                row = result.fetchone()
+
+                if bytes(row[row_name]) != bytes(test_string, "UTF-8"):
+                    self.fail(f"Table identifier {table_name}, column identifier {column_name}, matched (no plaintext in DB), {bytes(row['data'])} != {test_string}")
+
+                result = self.engine_raw.execute(f"SELECT {column_name} FROM {table_name} WHERE id={id};")
+                row = result.fetchone()
+
+                if bytes(row[row_name]) != bytes(test_string, "UTF-8"):
+                    self.fail(f"Table identifier {table_name}, column identifier {column_name}, matched (no plaintext in DB), {bytes(row['data'])} != {test_string}")
+
+        # insert a record
+        self.engine1.execute(f"INSERT INTO {table_name} (id, {column_name}) VALUES ({id}, '{test_string}');")
+        check()
+
+        # update a record
+        self.engine1.execute(f"UPDATE {table_name} SET {column_name}='{test_string}' WHERE id={id};")
+        check()
+
+    def testLowerConfigLowerQuery(self):
+        # table should match, lowercase config identifier == lowercase SQL identifier
+        # column should only match in quoted "DATA" case
+        self.runTestCase("lowercase_table", False, "data", False, False)
```

Comment:
when read such functions it is hard to understand what means these booleans and to not forget them. Can you use named arguments or named values:
```
QUOTED, NOT_QUOTED = (True, False)
SHOULD_MATCH, SHOULD_NOT_MATCH = (True, False)
self.runTestCase("lowercase_table", NOT_QUOTED, "data", QUOTED, SHOULD_NOT_MATCH)

```

Reply:
Sounds good, but... should I add these constants in all test methods? Or add to test class and access with `self.`? Both methods don't sound like a best solution. Making global constants would be even worse.

Reply:
it is okay to copy-paste it to each method that use it